osquery/events/windows/evtsubscription.h

```diff
@@ -0,0 +1,49 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <osquery/utils/status/status.h>
+
+#include <windows.h>
+#include <winevt.h>
+
+namespace osquery {
+DWORD WINAPI EvtSubscriptionCallbackDispatcher(
+    EVT_SUBSCRIBE_NOTIFY_ACTION action, PVOID context, EVT_HANDLE event);
+
+class EvtSubscription final {
+ public:
+  using Ref = std::unique_ptr<EvtSubscription>;
+  using Event = std::wstring;
+  using EventList = std::vector<Event>;
+
+  static Status create(Ref& obj, const std::string& channel);
+  ~EvtSubscription();
+
+  const std::string channel() const;
+  EventList getEvents();
+
+  EvtSubscription(const EvtSubscription&) = delete;
+  EvtSubscription& operator=(const EvtSubscription&) = delete;
+
+ private:
+  struct PrivateData;
+  std::unique_ptr<PrivateData> d;
```

Comment:
We typically call private member variables with an `_` suffix, so `d_`. But I recognize the PIMPL pattern and we can borrow from that pattern's recommendation if you have a strong opinion to do so.